### PR TITLE
Fix and unify internal links.

### PR DESCRIPTION
**Pull request by ycopin**
Internal links in specutils.analysis documentation are wrong for the Template Comparison section, and inconsistent within page.

### CODE REVIEW

**Comment by pllim** [comment]
Thanks! I think you missed one. 😸 

`/docs/analysis.rst:198:py:obj reference target not found: specutils.analysis.template_comparison.template_match`

**Comment by rosteen** [comment]
It doesn't look like one was missed, it looks like that path is not rendering correctly for some other reason. I'm not sure why after looking at it for a couple minutes, hm.

File: docs/analysis.rst [hunk 1]
```diff
@@ -195,13 +195,13 @@ Each of the width analysis functions are applied to this spectrum below:
...
 
-The ~`specutils.analysis.template_comparison.template_match` function takes an
+The `~specutils.analysis.template_comparison.template_match` function takes an
 observed spectrum and ``n`` template spectra and returns the best template that
 matches the observed spectrum via chi-square minimization.
 
 If the redshift is known, the user can set that for the ``redshift`` parameter
 and then run the
-~`specutils.analysis.template_comparison.template_match` function.
+`~specutils.analysis.template_comparison.template_match` function.
```
Comment (rosteen) [inline docs/analysis.rst:204]:
```suggestion
`~specutils.analysis.template_match` function.
```

File: docs/analysis.rst [hunk 2]
```diff
@@ -218,7 +218,7 @@ values to use. As an example, a simple linear grid can be built with:
 
     >>> rs_values = np.arange(1., 3.25, 0.25)
 
-The ~`specutils.analysis.template_comparison.template_match` function will then:
+The `~specutils.analysis.template_comparison.template_match` function will then:
```
Comment (rosteen) [inline docs/analysis.rst:221]:
```suggestion
The `~specutils.analysis.template_match` function will then:
```

**Review by rosteen — APPROVED** [review]
Thanks!

**Pull request merged**